### PR TITLE
Add sphinx-codeautolink/-copybutton to documentation and fix numpy docstrings

### PR DESCRIPTION
**Pull request by Zeitsperre**
### Pull Request Checklist:
- [ ] This PR addresses an already opened issue (for bug fixes / features)
    - This PR fixes #xyz
- [x] Tests for the changes have been added (for bug fixes / features)
  - [x] (If applicable) Documentation has been added / updated (for bug fixes / features)
- [x] HISTORY.rst has been updated (with summary of main changes)
  - [x] Link to issue (:issue:`number`) and pull request (:pull:`number`) has been added
- [x] The relevant author information has been added to `AUTHORS.rst` and `.zenodo.json`

### What kind of change does this PR introduce?

* Adds the `sphinx-codeautolink` extension to the docs
* Adds the `sphinx-copybutton` extension to the docs
* Converts docstrings skipped from Doctest into python code blocks
* Reorders bibliography to fix nesting problem
* Fixes several docstrings to adhere closer to the standard (https://numpydoc.readthedocs.io/en/latest/format.html)

### Does this PR introduce a breaking change?

No.

### Other information:

This PR also creates a series of session-scope `{variable}_dataset` accessors in the doctest namespace, as well as a `path_to_atmos_file` object. These can be used for running doctests on all variables made in the `atmosds` file.

https://github.com/felix-hilden/sphinx-codeautolink

I've opened an issue to see if the author is interested in contributing this package to `conda-forge` (see: https://github.com/felix-hilden/sphinx-codeautolink/issues/111). This package would be immensely useful in the documentation of many of our projects.

### CODE REVIEW

**Comment by Zeitsperre** [comment]
`sphinx-copybutton` works! 
image

File: xclim/core/options.py [hunk 1]
```diff
@@ -141,44 +141,43 @@ def run_check(*args, **kwargs):
...
-      How to check for missing data and flag computed indicators.
-      Default available methods are "any", "wmo", "pct", "at_least_n" and "skip".
-      Missing method can be registered through the `xclim.core.options.register_missing_method` decorator.
-      Default: ``'any'``
-    - ``missing_options``:
-      Dictionary of options to pass to the missing method. Keys must the name of
-      missing method and values must be mappings from option names to values.
-    - ``run_length_ufunc``:
+    Attributes
+    ----------
```
Comment (aulemahal) [inline xclim/core/options.py:145]: Attributes? Does this trigger something in the doc formatting?

If no, I would suggest something more precise like:
```suggestion
    Options
    -------
```

**Review by aulemahal — APPROVED** [review]
Nice! :mage_man: :magic_wand:

File: xclim/core/options.py (hunk 1 above)
Comment (Zeitsperre) [reply xclim/core/options.py:145]: Allegedly, this is how we're supposed to document classes: https://numpydoc.readthedocs.io/en/latest/format.html#class-docstring

**Pull request merged**